odoo/addons/base/tests/test_res_lang.py

```diff
@@ -40,3 +41,14 @@ def test_00_intersperse(self):
         assert intersperse("abc12", [3], '.') == ('abc12', 0)
         assert intersperse("abc12", [2], '.') == ('abc12', 0)
         assert intersperse("abc12", [1], '.') == ('abc1.2', 1)
+
+    def test_inactive_users_lang_deactivation(self):
+        resUser, resLang = self.env['res.users'], self.env['res.lang']
+
+        resLang._activate_lang('en_GB')
+        inactive_users_lang = self.env['res.users'].search([('active', '=', False)]).mapped('lang')
+        resUser.search([('lang', 'in', inactive_users_lang)]).write({'lang': 'en_GB'})
+        self.env['res.partner'].search([('lang', 'in', inactive_users_lang)]).write({'lang': 'en_GB'})
+
+        with self.assertRaises(UserError), self.cr.savepoint():
+            resLang.search([('code', 'in', inactive_users_lang)]).active = False
```

Comment:
This code is more complex than necessary, I don't see the point of all those calls to `search`.  Let's make specific records for what we need:
```suggestion
    def test_inactive_users_lang_deactivation(self):
        # retrieve language en_GB, and make it active
        language = self.env['res.lang']._activate_lang('en_GB')
        
        # assign it to an inactive (new) user
        user = self.env['res.users'].create({
            'name': 'Foo',
            'login': 'foo@example.com',
            'lang': 'en_GB',
            'active': False,
        })
        
        # make sure it is only used by that user
        self.assertEqual(user.lang, 'en_GB')
        self.assertEqual(self.env['res.users'].with_context(active_test=False).search_count([('lang', '=', 'en_GB')]), 1)

        with self.assertRaises(UserError):
            language.active = False
```